### PR TITLE
DATA-1889 - Add ConfigureDatabaseUser to the API

### PR DESCRIPTION
**Pull request by dmhilly**

### CODE REVIEW

**Review by agreenb — APPROVED** [review]
LGTM after one nit!

File: proto/viam/app/data/v1/data.proto [hunk 1]
```diff
@@ -55,6 +55,10 @@ service DataService {
   // BoundingBoxLabelsByFilter gets all string labels for bounding boxes from data based on given filter.
   rpc BoundingBoxLabelsByFilter(BoundingBoxLabelsByFilterRequest) returns (BoundingBoxLabelsByFilterResponse);
 
+  // ConfigureDatabaseUser configures a database user for the Viam organization. It can also be used to
+  // reset the password of an existing database user.
```
Comment (kaywux) [inline proto/viam/app/data/v1/data.proto:59]: Can we add some context to the comment to distinguish this from other mongo dbs? Can be something as simple as "... configures a _federated_ database user ...."

**Review by kaywux — CHANGES_REQUESTED** [review]
One tiny request for a comment, otherwise looks great!

File: proto/viam/app/data/v1/data.proto (hunk 1 above)
Comment (dmhilly) [reply proto/viam/app/data/v1/data.proto]: Yes good point, this isn't very clear. Added more to the comment!

**Review by kaywux — APPROVED** [review]
LGTM, thanks Devin!

**Review by kaywux — APPROVED** [review]

**Pull request merged**